### PR TITLE
Feature/sci strategic ros2

### PR DESCRIPTION
**Pull request by MishkaMN**
# PR Details
## Description

This PR includes updates to port the sci_strategic_plugin from ROS1 to ROS2.

## Related Issue
https://github.com/usdot-fhwa-stol/carma-platform/issues/1906

## How Has This Been Tested?

All unit tests pass, and local integration tests have been conducted.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: sci_strategic_plugin/src/sci_strategic_plugin.cpp [hunk 1]
```diff
@@ -810,11 +899,26 @@ cav_msgs::Maneuver SCIStrategicPlugin::composeIntersectionTransitMessage(double
...
+  RCLCPP_INFO_STREAM(rclcpp::get_logger("sci_strategic_plugin"), "Creating IntersectionTransitManeuver start dist: " << start_dist << " end dist: " << end_dist
                                                                       << " From lanelet: " << starting_lane_id
                                                                       << " to lanelet: " << ending_lane_id);
 
   return maneuver_msg;
 }
 
+bool SCIStrategicPlugin::get_availability()
+{
+  return true;
```
Comment (adev4a) [inline sci_strategic_plugin/src/sci_strategic_plugin.cpp:911]: What is this meant for?

File: sci_strategic_plugin/launch/sci_strategic_plugin.launch.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            
+            # Launch the core node(s)
+            ComposableNode(
+                    package='sci_strategic_plugin',
+                    plugin='sci_strategic_plugin::SCIStrategicPlugin',
+                    name='sci_strategic_plugin_node',
+                    extra_arguments=[
+                        {'use_intra_process_comms': True},
+                        {'--log-level' : log_level }
+                    ],
```
Comment (adev4a) [inline sci_strategic_plugin/launch/sci_strategic_plugin.launch.py:54]: We need to load the parameter file here

File: sci_strategic_plugin/src/sci_strategic_plugin.cpp [hunk 3]
```diff
@@ -33,90 +33,177 @@
...
+  config_.intersection_transit_plugin_name = declare_parameter<std::string>("intersection_transit_plugin_name", config_.intersection_transit_plugin_name);
+  config_.vehicle_id = declare_parameter<std::string>("vehicle_id", config_.vehicle_id);
+  config_.veh_length = declare_parameter<double>("vehicle_length", config_.veh_length);
+  config_.vehicle_decel_limit = declare_parameter<double>("vehicle_deceleration_limit", config_.vehicle_decel_limit);
+  config_.vehicle_accel_limit = declare_parameter<double>("vehicle_acceleration_limit", config_.vehicle_accel_limit);
+
 };
 
-cav_msgs::Plugin SCIStrategicPlugin::getDiscoveryMsg() const
+carma_ros2_utils::CallbackReturn SCIStrategicPlugin::on_configure_plugin()
```
Comment (adev4a) [inline sci_strategic_plugin/src/sci_strategic_plugin.cpp:90]: Can we add the parameter update callback as well for the parameters not coming from vehicle configuration?

File: sci_strategic_plugin/src/sci_strategic_plugin.cpp (hunk 1 above)
Comment (MishkaMN) [reply sci_strategic_plugin/src/sci_strategic_plugin.cpp:911]: This is an override of base virtual class function. It has to be true by default i think, but subsystem_controllers are capable of turning it false when it is needed to turn off.
https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/carma_guidance_plugins/include/carma_guidance_plugins/plugin_base_node.hpp#L139

File: sci_strategic_plugin/launch/sci_strategic_plugin.launch.py (hunk 2 above)
Comment (MishkaMN) [reply sci_strategic_plugin/launch/sci_strategic_plugin.launch.py:54]: added parameter file

File: sci_strategic_plugin/src/sci_strategic_plugin.cpp (hunk 3 above)
Comment (MishkaMN) [reply sci_strategic_plugin/src/sci_strategic_plugin.cpp:91]: added

File: sci_strategic_plugin/src/sci_strategic_plugin.cpp [hunk 4]
```diff
@@ -33,90 +33,206 @@
...
+    {"vehicle_decel_limit_multiplier", config_.vehicle_decel_limit_multiplier},
+    {"vehicle_accel_limit_multiplier", config_.vehicle_accel_limit_multiplier},
+    {"stop_line_buffer", config_.stop_line_buffer},
+    {"delta_t", config_.delta_t},
+    {"min_gap", config_.min_gap},
+    {"reaction_time", config_.reaction_time},
+    {"intersection_exit_zone_length", config_.intersection_exit_zone_length}
+  }, parameters); // vehicle_acceleration_limit not updated as it's global param
+
+  auto error_string = update_params<std::string>({
```
Comment (adev4a) [inline sci_strategic_plugin/src/sci_strategic_plugin.cpp:150]: We probably dont want to be able to change the plugins while the node is up? I think we can remove all 3 of them here

Comment (MishkaMN) [reply sci_strategic_plugin/src/sci_strategic_plugin.cpp:150]: good point

**Review by adev4a — APPROVED** [review]

**Pull request merged**